RoutesApp_ios/modules/lineList/views/controllers/RouteListViewController.swift

```diff
@@ -0,0 +1,85 @@
+    let lineRouteViewController = LineRouteViewController()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        lineListTableView.dataSource = self
+        lineListTableView.delegate = self
+        setupNavigationBar()
+        setIcon()
+        initViewModel()
+        routeDetailViewModel.getLines {
+            self.lineListTableView.reloadData()
+        }
+        lineListTableView.register(UINib.init(nibName: ConstantVariables.routeListCell,
+        bundle: nil), forCellReuseIdentifier: ConstantVariables.routeListCell)
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        self.lineListTableView.reloadData()
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+    func initViewModel() {
+        routeDetailViewModel.fecthedLineRoute = { [weak self] () in
+            guard let lineRouteList = self?.routeDetailViewModel.lineRouteList else { return }
+            if lineRouteList.count > 1 {
+                self?.lineRouteViewController.lineRouteList = lineRouteList
+                let screenSize: CGRect = UIScreen.main.bounds
+                let width = screenSize.width - 30
+                let height = screenSize.height / 4
+                let popupVC = PopupViewController(contentController: self!.lineRouteViewController, popupWidth: width, popupHeight: height)
+                popupVC.cornerRadius = 10
+                self?.present(popupVC, animated: true)
+            } else if lineRouteList.count == 1 {
+                let linePath = self?.routeDetailViewModel.convertToLinePath(lineRouteInfo: lineRouteList[0])
+                let routeMapViewController = RouteMapViewController()
+                routeMapViewController.linePath = linePath
+                self?.present(routeMapViewController, animated: false)
+            }
+        }
+    }
+    func setupNavigationBar() {
+        navigationItem.title = String.localizeString(localizedString: "Lines")
+        self.navigationItem.largeTitleDisplayMode = .always
+        navigationController?.navigationBar.barTintColor = UIColor(named: ConstantVariables.primaryColor)
+        let routeListVC = RouteListViewController()
+        let searchController = UISearchController(searchResultsController: routeListVC)
+        navigationItem.searchController = searchController
+        let colorValue = [NSAttributedString.Key.foregroundColor: UIColor.white]
+        UITextField.appearance(whenContainedInInstancesOf: [UISearchBar.self]).defaultTextAttributes = colorValue
+        UINavigationBar.appearance().isTranslucent = false
+    }
+
+    func setIcon() {
+        let filterIcon = UIImage(named: ConstantVariables.filterIcon)?.withRenderingMode(.alwaysOriginal)
+        let filterButton = UIBarButtonItem(image: filterIcon, style: .plain, target: self, action: #selector(showFilterPopUp))
+        navigationItem.rightBarButtonItem = filterButton
+    }
+
+    @objc func showFilterPopUp() {
```

Comment:
is this empty function necessary?

Reply:
I don't know, that was already there, i think its for filter functionality

Pd: just in case, everything is green because I changed the name of the class

Reply:
That function is just there because there's a #selector when setting the icon, It was empty so the developer in charge can directly work with it, I think the one doing it it's Israel on his pull request with the filter and searchbar